transaction-status/src/parse_token.rs

```diff
@@ -0,0 +1,550 @@
+        assert_eq!(
+            parse_token(&compiled_instruction, &keys[0..3]).unwrap(),
+            json!({
+                "type": "initializeAccount",
+                "account": keys[0].to_string(),
+                "mint": keys[1].to_string(),
+                "owner": keys[2].to_string(),
+            })
+        );
+
+        // Test InitializeMultisig
+        let initialize_multisig_ix = initialize_multisig(
+            &spl_token_v1_0::id(),
+            &convert_pubkey(keys[0]),
+            &[
+                &convert_pubkey(keys[1]),
+                &convert_pubkey(keys[2]),
+                &convert_pubkey(keys[3]),
+            ],
+            2,
+        )
+        .unwrap();
+        let message = Message::new(&[initialize_multisig_ix], None);
+        let compiled_instruction = convert_compiled_instruction(&message.instructions[0]);
+        assert_eq!(
+            parse_token(&compiled_instruction, &keys).unwrap(),
+            json!({
+                "type": "initializeMultisig",
+                "multisig": keys[0].to_string(),
+                "m": 2,
+                "signers": keys[1..4].iter().map(|key| key.to_string()).collect::<Vec<String>>(),
+            })
+        );
+
+        // Test Transfer, incl multisig
+        let transfer_ix = transfer(
+            &spl_token_v1_0::id(),
+            &convert_pubkey(keys[1]),
+            &convert_pubkey(keys[2]),
+            &convert_pubkey(keys[0]),
+            &[],
+            42,
+        )
+        .unwrap();
+        let message = Message::new(&[transfer_ix], None);
+        let compiled_instruction = convert_compiled_instruction(&message.instructions[0]);
+        assert_eq!(
+            parse_token(&compiled_instruction, &keys).unwrap(),
+            json!({
+                "type": "transfer",
+                "source": keys[1].to_string(),
+                "destination": keys[2].to_string(),
+                "authority": keys[0].to_string(),
+                "amount": 42,
+            })
+        );
+
+        let transfer_ix = transfer(
+            &spl_token_v1_0::id(),
+            &convert_pubkey(keys[2]),
+            &convert_pubkey(keys[3]),
+            &convert_pubkey(keys[4]),
+            &[&convert_pubkey(keys[0]), &convert_pubkey(keys[1])],
+            42,
+        )
+        .unwrap();
+        let message = Message::new(&[transfer_ix], None);
+        let compiled_instruction = convert_compiled_instruction(&message.instructions[0]);
+        assert_eq!(
+            parse_token(&compiled_instruction, &keys).unwrap(),
+            json!({
+                "type": "transfer",
+                "source": keys[2].to_string(),
+                "destination": keys[3].to_string(),
+                "multisigAuthority": keys[4].to_string(),
+                "signers": keys[0..2].iter().map(|key| key.to_string()).collect::<Vec<String>>(),
+                "amount": 42,
+            })
+        );
+
+        // Test Approve, incl multisig
```

Comment:
Shall we re-release token now that this is fixed in: https://github.com/solana-labs/solana-program-library/pull/164

Reply:
Sounds good. I'll publish a new patch version tonight or tomorrow am.